lib/perl/Genome/Model/Tools/SpeedSeq/AlignBase.pm

```diff
@@ -0,0 +1,141 @@
+            is_optional => 1,
+        },
+        temp_directory => {
+            is => 'Text',
+            doc => 'temp directory',
+            example_values => ['./output_prefix.XXXXXXXXXXXX'],
+            is_optional => 1,
+        },
+        insert_size => {
+            is => 'Text',
+            doc => 'specify the mean, standard deviation (10% of the mean if absent), max(4 sigma from the mean if absent) and min of the insert size distribution.  FR orientation only. Inferred by default.',
+            is_optional => 1,
+            example_values => ['FLOAT[,FLOAT[,INT[,INT]]]'],
+        },
+        include_duplicates =>{
+            is => 'Boolean',
+            doc => 'include duplicates in splitters and discordants',
+            is_optional => 1,
+        },
+        maximum_split_alignments => {
+            is => 'Integer',
+            doc => 'maximum number of split alignments for a read to be included in splitter file',
+            is_optional => 1,
+            example_values => ['2'],
+        },
+        minimum_non_overlapping_bp => {
+            is => 'Integer',
+            doc => 'minimum non-overlapping base pairs between two alignments for a read to be included in splitter file',
+            is_optional => 1,
+            example_values => ['20'],
+        },
+        sort_memory => {
+            is => 'Integer',
+            doc => 'amount of memory in GB to be used for sorting',
+            is_optional => 1,
+            example_values => ['20'],
+        },
+    ],
+    has_optional_output => [
+        output_files => {
+            is_many => 1,
+        },
+    ],
+};
+
+sub _output_extensions {
+    my $class = shift;
+    return qw/
+                 .bam
+                 .bam.bai
+                 .discordants.bam
+                 .discordants.bam.bai
+                 .splitters.bam
+                 .splitters.bam.bai
+             /;
+}
+
+sub _resolve_options_string {
+    my $self = shift;
+    my $params = shift;
+
+    unless ($params) {
+        $params = '';
+    }
+    if ($self->output_prefix) {
+        $params .= ' -o '. $self->output_prefix;
+    }
+    if ($self->threads) {
+        $params .= ' -t '. $self->threads;
+    }
+    if ($self->temp_directory) {
+        $params .= ' -T '. $self->temp_directory;
+    }
+    if ($self->insert_size) {
+        $params .= ' -I '. $self->insert_size;
+    }
+    if ($self->include_duplicates) {
+        $params .= ' -i ';
+    }
+    if ($self->maximum_split_alignments) {
+        $params .= ' -c '. $self->maximum_split_alignments;
+    }
+    if ($self->minimum_non_overlapping_bp) {
+        $params .= ' -m '. $self->minimum_non_overlapping_bp;
+    }
+    if ($self->sort_memory) {
+        $params .= ' -M '. $self->sort_memory;
+    }
+    if ($self->config_file) {
+        $params .= ' -K '. $self->config_file;
+    }
+    if ($self->verbose) {
+        $params .= ' -v ';
+    }
+    return $params;
+}
+
+sub _resolve_output_file_paths {
```

Comment:
This could probably also be moved to SpeedSeq base, although this refactoring isn't critical until we add more of the speedseq tool wrappers.